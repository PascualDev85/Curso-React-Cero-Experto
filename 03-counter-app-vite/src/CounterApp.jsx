
import PropTypes from 'prop-types'
import { useState } from 'react'

export const CounterApp = ({value}) => {
    
    console.log('render');
    // cuando cambia el estado el componente se vuelve ejecutar
    // si hay una llamada asincrona se vuelve a disparar con el useEffect

    const [counter, setCounter] = useState(value);
    
    const handleAdd = () => {
        // console.log(e);
        
        //VALIDA LAS DOS MANERAS (handleAdd y handleSubtract)
        setCounter((c) => c + 1)
        
    }

    const handleSubtract = () => {
        setCounter(counter - 1)
    }

    // const handleReset = () => {
    //     setCounter(value);
    // }

    return (
        <>
        <h1>CounterApp</h1>
        <h2> {counter} </h2>
        <button onClick={handleAdd}> + 1</button>
        <button onClick={() => setCounter(value) }> Reset</button>
        <button aria-label="btn-reset" onClick={handleSubtract}> - 1 </button>
        </>
  )
}

CounterApp.protoTypes = {
    value: PropTypes.number.isRequired
}
