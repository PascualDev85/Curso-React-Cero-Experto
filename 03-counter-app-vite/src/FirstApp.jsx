import PropTypes  from 'prop-types';


// Regla si no se cambia este valor se saca fuera y no tiene relación con algún 
//Hook o depencia


// const newMessage = {
//     message: 'Hola Mundo',
//     nombre: 'David'
// };

const getREsult = (a, b) => {
    return a + b;
}

export const FirstApp = ( {title, subTitle, name} ) => {
    // LAS PROPOS SE PUEDEN PASAR DEL COMPONENTE PADRE A SUS HIJOS
    // console.log(props);
  
    
  return (
    <>
    {/* Agrupadro de etiquets de jsx */}
        <h1 data-testid="test-title">{title}</h1>
        {/* <code>{JSON.stringify(newMessage)}</code> */}
        {/* Para poder imprimir un objeto */}
        <hr />
        <p>{subTitle}</p>
        <p>{subTitle}</p>
        <p>{name}</p> 
    </>
  )
}

// SE COLOCAN AL FINAL
// sirven para poder definr por defecto las props
FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.string,
}

// Los defaultProps van a entrar antes que los propTypes por tanto no saldrá error, a no ser que se envie props diferentes tipos del que se marcó.
FirstApp.defaultProps = {
    name: 'David Pascual',
    title: 'No hay título',
    subTitle: 'No hay subTitle',
}