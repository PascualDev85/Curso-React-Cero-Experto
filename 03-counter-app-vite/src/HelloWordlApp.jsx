//  function App() {
//     return (
//       //document.createElement...
//       <div>
//         <h1>Hello, world!!!!</h1>
//       </div>
//     );
//   }

// export default App;


export const HelloWordlApp = () => {
  return (
    <h1>Hello Wordl App</h1>
  );
}
