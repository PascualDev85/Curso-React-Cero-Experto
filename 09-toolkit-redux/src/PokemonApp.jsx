import { useEffect } from "react";
import { useDispatch, useSelector } from "react-redux";
import { getPokemons } from "./store/slices/pokemon";

export const PokemonApp = () => {
  const {
    isLoading,
    pokemons = [],
    page,
  } = useSelector((state) => state.pokemon);
  const dispatch = useDispatch();

  console.log(pokemons);

  useEffect(() => {
    dispatch(getPokemons());
  }, []);

  return (
    <>
      <h1>Pokemon App</h1>
      <hr />
      <span>Loading: {!isLoading ? "True" : "False"}</span>
      <ul>
        {pokemons.map(({ name }) => (
          <li key={name}>{name}</li>
        ))}
      </ul>
      <p>{page}</p>
      <button
        disabled={isLoading || page === 1}
        onClick={() => dispatch(getPokemons(page - 2))}
      >
        Previous
      </button>
      <button
        disabled={isLoading || page === 1}
        onClick={() => dispatch(getPokemons(page - page))}
      >
        Home
      </button>
      <button disabled={isLoading} onClick={() => dispatch(getPokemons(page))}>
        Next
      </button>
    </>
  );
};
