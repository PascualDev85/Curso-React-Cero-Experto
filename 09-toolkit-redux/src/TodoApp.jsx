import { useState } from "react";
import { useGetTodoByIdQuery, useGetTodosQuery } from "./store/apis";

export const TodoApp = () => {
  //   const { data: todos = [], isLoading } = useGetTodosQuery();
  //   console.log(algo);

  const [todoId, setTodoId] = useState(1);
  const { data: todo, isLoading } = useGetTodoByIdQuery(todoId);
  //   console.log(todo);

  const nextTodo = () => {
    setTodoId(todoId + 1);
  };

  const prevTodo = () => {
    if (todoId === 1) return false;
    setTodoId(todoId - 1);
  };

  return (
    <>
      <h1>Todos - RFT Query</h1>
      <hr />
      <h4>isLoading: {isLoading ? "True" : "False"}</h4>

      <pre> {JSON.stringify(todo)} </pre>

      {/* <ul>
        {todos.map((todo) => (
          <li key={todo.id}>
            <strong>{todo.completed ? "DONE " : "Pending "}</strong>
            {todo.title}
          </li>
        ))}
      </ul> */}
      <button onClick={prevTodo}>Prev Todo</button>
      <button onClick={nextTodo}>Next Todo</button>
    </>
  );
};
