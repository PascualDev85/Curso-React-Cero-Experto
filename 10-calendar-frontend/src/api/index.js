// ponemos as porque tiene una exportaci√≥n por defecto
export { default as calendarApi } from "./calendarApi";
