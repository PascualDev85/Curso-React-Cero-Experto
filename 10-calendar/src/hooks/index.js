export * from "./useEventStyleCalendar";
export * from "./useUiStore";
export * from "./useCalendarStore";
