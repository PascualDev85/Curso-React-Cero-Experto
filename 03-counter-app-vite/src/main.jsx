import React from 'react';
import ReactDOM from 'react-dom/client';
// import { HelloWordlApp }  from './HelloWordlApp';
// import { FirstApp } from './FirstApp';
import { CounterApp } from './CounterApp';

import './index.css';


ReactDOM.createRoot(document.getElementById('root')).render(
<React.StrictMode>
{/* <HelloWordlApp /> */}
{/* <FirstApp title="Hola soy, Goku"/> */}
{/* // si queremos mandar un boolean bastaría con pasar solo por ej: title sólo */}
{/* /> */}
<CounterApp value={10} />
</React.StrictMode>
);