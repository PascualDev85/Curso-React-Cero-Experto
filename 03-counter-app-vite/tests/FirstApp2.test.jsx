import { render, screen } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";

describe('Pruebas en <FirstApp />', () => {
    const title = "Hola, Soy Goku";
    const subTitle = "Soy el subtitulo";

    test('debe de hacer match con el snapshot', () => {
        const { container } = render(<FirstApp title={title} />);
        expect(container).toMatchSnapshot();
    })
   
    test('debe de mostar el mensaje "Hola, soy Goku"', () => { 
        render(<FirstApp title={title} />)
        expect(screen.getByText(title)).toBeTruthy();
    
     })

     test('debe mostar el titulo en un h1', () => {
        render(<FirstApp title={title} />)
        expect(screen.getByRole('heading', {level: 1}).innerHTML).toContain(title);
     });

     test('debe mostrar el subtitulo', () => { 

        render(
        <FirstApp 
            title={title} 
            subTitle={subTitle}
        />)
        expect(screen.getAllByText(subTitle).length).toBe(2);
      })

})