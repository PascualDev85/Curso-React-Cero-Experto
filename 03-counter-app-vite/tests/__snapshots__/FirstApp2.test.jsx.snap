// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> debe de hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola, Soy Goku
  </h1>
  <hr />
  <p>
    No hay subTitle
  </p>
  <p>
    No hay subTitle
  </p>
  <p>
    David Pascual
  </p>
</div>
`;
